include/linker/devicetree_regions.h

```diff
@@ -7,6 +7,36 @@
  * Generate memory regions from devicetree nodes.
  */
 
+/**
+ * @brief Get the linker memory-region name
+ *
+ * This attempts to use the zephyr,linker-region property, falling back
+ * to the node path if it doesn't exist.
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     sram1: memory@2000000 {
+ *                         zephyr,linker-region = "MY_NAME";
+ *                     };
+ *                     sram2: memory@2001000 { ... };
+ *             };
+ *     };
+ *
+ * Example usage:
+ *
+ *    LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram1)) // "MY_NAME"
+ *    LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram2)) // "/soc/memory@2001000"
+ *
+ * @param node_id node identifier
+ * @return the name of the linker region the node will generate
+ */
+#define LINKER_DT_NODE_REGION_NAME(node_id)			     \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, zephyr_linker_region), \
+		    (DT_PROP(node_id, zephyr_linker_region)),	     \
+		    (DT_NODE_PATH(node_id)))
```

Comment:
This could be

```
DT_PROP_OR(node_id, zephyr_linker_region, DT_NODE_PATH(node_id))
```

Not a blocking comment, just an FYI.

Reply:
That is a much cleaner implementation, done